Pepperdash Core/Pepperdash Core/Comm/GenericSecureTcpIpClient.cs

```diff
@@ -0,0 +1,1048 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Crestron.SimplSharp;
+using Crestron.SimplSharp.CrestronSockets;
+
+namespace PepperDash.Core
+{
+    public class GenericSecureTcpIpClient : Device, ISocketStatusWithStreamDebugging, IAutoReconnect
+    {
+        private const string SplusKey = "Uninitialized Secure Tcp Client";
+        /// <summary>
+        /// Stream debugging 
+        /// </summary>
+        public CommunicationStreamDebugging StreamDebugging { get; private set; }
+
+        /// <summary>
+        /// Fires when data is received from the server and returns it as a Byte array
+        /// </summary>
+        public event EventHandler<GenericCommMethodReceiveBytesArgs> BytesReceived;
+
+        /// <summary>
+        /// Fires when data is received from the server and returns it as text
+        /// </summary>
+        public event EventHandler<GenericCommMethodReceiveTextArgs> TextReceived;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        //public event GenericSocketStatusChangeEventDelegate SocketStatusChange;		
+        public event EventHandler<GenericTcpServerSocketStatusChangeEventArgs> ConnectionChange;
+
+        #region GenricTcpIpClient properties
+
+        private string _hostname;
+
+        /// <summary>
+        /// Address of server
+        /// </summary>
+        public string Hostname
+        {
+            get { return _hostname; }
+            set
+            {
+                _hostname = value;
+                if (Client != null)
+                {
+                    Client.AddressClientConnectedTo = _hostname;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Port on server
+        /// </summary>
+        public int Port { get; set; }
+
+        /// <summary>
+        /// S+ helper
+        /// </summary>
+        public ushort UPort
+        {
+            get { return Convert.ToUInt16(Port); }
+            set { Port = Convert.ToInt32(value); }
+        }
+
+        /// <summary>
+        /// Defaults to 2000
+        /// </summary>
+        public int BufferSize { get; set; }
+
+        /// <summary>
+        /// Internal secure client
+        /// </summary>
+        public SecureTCPClient Client { get; private set; }
```

Comment:
This should be private and a field.
```
private SecureTCPClient _client;
```

As we're wrapping the client and adding additional functionality, we don't want it to be accessible from outside of this class.

Reply:
Changed client to private and renamed to reflect a private instance.